hs_core/urls.py

```diff
@@ -31,10 +34,12 @@
     url(r'^resource/(?P<pk>[A-z0-9]+)/files/(?P<filename>[^/]+)/$',
```

Comment:
@pkdash @hyi This does not solve the file path problem. 
Consider `^resource/(P<pk>{A-z0-9]+)/files/(?P<filename>[.a-zA-Z0-9 -_/]+)/$`
What this does: a file path consists of a string of characters a-z, A-Z, 0-9, ' ', '-', '_', and '/' that ends with a '/'.
In Linux the following characters are usually not used in filenames: 

* ';', '/', '\' are shell escapes. 
* '*' is the shell glob character. 
* ',' is the glob separator 
* '$' references a shell variable. 
* '&' puts a job into the background. 
* '%' references a job. 
* '!' references a previous command. 
* '()' defines a shell array. 
* '{}' invokes special variable manipulations. 
* '`' invokes a shell command. 

For security reasons, we also need to disallow '/../' and '/./' and '//'. The regular expression engine allows me to accomplish this. 

Or, if you want, we could simply accept everything and then validate it later. E.g., `^resource/(P<pk>{A-z0-9]+)/files/(?P<filename>.+)/$` I.e., accept anything that ends with a / delimiter.